### PR TITLE
Added note regarding workflow "entered" event being dispatched on creation

### LINKED ISSUE #12732: [Workflow] Document the new event 'entered' triggered on workflow enter for first time
This PR was merged a year ago but wasn't documented.
https://github.com/symfony/symfony/pull/29145

I will try to submit a PR this weekend.

### PR DESCRIPTION
**Pull request by florentdestremau**
Fixes #12732

### CODE REVIEW

File: workflow.rst [hunk 1]
```diff
@@ -337,7 +337,8 @@ order:
 .. note::
 
     The leaving and entering events are triggered even for transitions that stay
-    in same place.
+    in same place. In addition to that, the following entering events are triggered upon the
+    workflow creation: `workflow.entered` and `workflow.[worflow name].entered`.
```
Comment (noniagriconomie) [inline workflow.rst]: Workflow creation is weird 

Why not : Following events are also dispatched when the subject entered the workflow for the first time

Comment (florentdestremau) [reply workflow.rst]: ...when the subject enters the workflow... ?

Comment (noniagriconomie) [reply workflow.rst]: I do not understand your answer

My first comment was in fact to remove the wording of « creation » as the workflow is not created neither create something 

That is why i proposed another wording, with the transition name « entered » and your idea of the « first » entrance :)

File: workflow.rst [hunk 2]
```diff
@@ -337,7 +337,8 @@ order:
 .. note::
 
     The leaving and entering events are triggered even for transitions that stay
-    in same place.
+    in same place. In addition to that, the following entering events are triggered upon the
```
Comment (OskarStark) [inline workflow.rst]: Please us double backticks instead of only single ones to highlight

**Review by OskarStark — COMMENTED** [review]
And please add a `.. versionadded:: 4.3` directive

**Review by OskarStark — APPROVED** [review]

**Comment by florentdestremau** [comment]
@OskarStark  so should I make a separate block for the whole sentence ?

File: workflow.rst (hunk 1 above)
Comment (florentdestremau) [reply workflow.rst]: Ok I get what you mean, updating.

**Comment by florentdestremau** [comment]
phew sorry with the rebase I got confused and didn't want to introduce unnecessary modifications. 
Is this good now @OskarStark and @noniagriconomie  ?

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
It took time, but here we go, this is in now. Thank you very much Florent.

**Review by noniagriconomie — APPROVED** [review]
👍

**Pull request merged**